src/functions/secrets.ts

```diff
@@ -0,0 +1,88 @@
+import { createSecret, getSecret, patchSecret, Secret } from "../gcp/secretManager";
+import { Options } from "../options";
+import { FirebaseError } from "../error";
+import { logWarning } from "../utils";
+import { promptOnce } from "../prompt";
+
+const FIREBASE_MANGED = "firebase-managed";
+
+/**
+ * Returns true if secret is managed by Firebase.
+ */
+function isFirebaseManaged(secret: Secret): boolean {
+  return Object.keys(secret.labels || []).includes(FIREBASE_MANGED);
+}
+
+/**
+ * Return labels to mark secret as managed by Firebase.
+ * @internal
+ */
+export function labels(): Record<string, string> {
+  return { [FIREBASE_MANGED]: "true" };
+}
+
+function transformKey(key: string): string {
+  return key
+    .replace("-", "_")
+    .replace(".", "_")
+    .replace(/([a-z])([A-Z])/g, "$1_$2")
+    .toUpperCase();
+}
+
+/**
+ * Validate and transform keys to match the convention recommended by Firebase.
+ */
+export function ensureValidKey(key: string, options: Options): string {
+  const transformedKey = transformKey(key);
+  if (transformedKey !== key) {
+    if (options.force) {
+      throw new FirebaseError("Secret key must be in UPPER_SNAKE_CASE.");
+    }
+    logWarning(
+      `By convention, secret key must be in UPPER_SNAKE_CASE. Using ${transformedKey} as key instead.`
```

Comment:
Personally, I would probably expect this is always throw an error instead of using a transformed key. Curious to why we want to change the key for the user.

Reply:
hmmm yeah that's a good call. I'll also add logic to reject secret names that uses reserved env keys specified in https://github.com/firebase/firebase-tools/blob/master/src/functions/env.ts#L12-L38.